frontend/sozisel-app/src/components/AuthRoute/AuthRoute.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { ReactElement, useEffect } from "react";
+import { Route, useHistory } from "react-router";
+
+import { useMeQuery } from "../../graphql";
+
+interface AuthRouteProps {
+  component?: React.ComponentType;
+  path: string;
+}
+
+export default function AuthRoute({
+  component,
+  path,
+}: AuthRouteProps): ReactElement {
+  const { error, loading } = useMeQuery();
+  const history = useHistory();
+
+  useEffect(() => {
+    if (error && error.message === "unauthorized") {
+      console.error(error);
+      history.push("/login");
+      localStorage.removeItem("token");
+    }
+  }, [error, history]);
+
+  if (!loading) {
```

Comment:
Overall I would return most important component in the last return as it is more readable. You could do the 
```js
if (loading || error) return <></>;

return <Route...
```